ts-fold.el

```diff
@@ -305,14 +304,12 @@ If the current node is not folded or not foldable, do nothing."
   "Fold all foldable syntax nodes in the buffer."
   (interactive)
   (ts-fold--ensure-ts
-    (let* ((node (tsc-root-node tree-sitter-tree))
-           (patterns (seq-mapcat (lambda (fold-range) `((,(car fold-range)) @name))
-                                 (alist-get major-mode ts-fold-range-alist)
-                                 'vector))
-           (query (tsc-make-query tree-sitter-language patterns))
-           (nodes-to-fold (tsc-query-captures query node #'ignore)))
+    (let* ((node (treesit-buffer-root-node))
+           (patterns (mapconcat (lambda (fold-range) (concat "(" (car fold-range) ") " "@name"))
+                                 (alist-get major-mode ts-fold-range-alist) " "))
+           (query (treesit-query-compile (treesit-node-language node) patterns))
+           (nodes-to-fold (treesit-query-capture node query nil nil t)))
```

Comment:
With `treesit.el` queries have to be strings, not s-expressions. So instead of seq-mapcat, we just concat everything into one big string.